### DISCUSSION TITLE
Code splitting

### DISCUSSION THREAD

**Discussion by Emiliano-Bucci** [chakra-ui/panda#1436] [General]
Hi! Is there a way to implement code splitting and reduce total bundle size? Or in case, would something like this will implemented? Thanks!

**Comment by astahmer** [comment]
I don't understand the question, do you mean : is there a way to split the CSS generated ? if that's the question, then the answer is yes there's a `panda cssgen` command that you can use like this `panda cssgen --outfile dist/styles.css`

you can also specify a glob of files that will override your config.include so you could use that command just to generate the CSS of a few components or pages

**Comment by Emiliano-Bucci** [comment]
@astahmer Yes, sorry, I meant that! I was thinking to a solution out of the box for next js, like splitting css by pages or something like that. I suppose it could be done manually though

**Reply by astahmer** [reply]
Panda doesn't transforms your source code, outside of the postcss entry file (that contains the `@layers`), so that can't be possible as of now since we don't have a dedicated bundler plugin for next (and vite etc) that would import a CSS file in your source code for each pages

anyway, you can indeed do it manually !

**Reply by Emiliano-Bucci** [reply]
@astahmer Yup! Could you please give me a short example of how could I achieve this? Thanks a lot!

**Reply by astahmer** [reply]
you can use it like this:

`pnpm panda cssgen "src/pages/**/page.tsx" -o ./path/to/generated/styles.css`

we're using [`fast-glob`](https://www.npmjs.com/package/fast-glob) if you need more documentation on the glob syntax

**Reply by Emiliano-Bucci** [reply]
@astahmer Thanks!

**Reply by Emiliano-Bucci** [reply]
@astahmer I've tried with few pages and Panda is generating the same big file for every page (with everything included), the only difference is that now is creating css chunks for every components, which is great. Could you please explain why is doing this and eventually how could I take advantage of this?

**Reply by astahmer** [reply]
if you split your CSS by page, that also means you'll need to have your tokens in each of those pages so I think it makes sense to still include everything

but yeah you have a point we should probably add a flag to `cssgen` so that you can skip the common stuff and just have the included files specific CSS, so you could cache that common

would that solve your use-case ?

**Reply by Emiliano-Bucci** [reply]
@astahmer Yes, I guess that we should generate global (common) stuff and specific stuff, so I will import common stuff in **_app.tsx** and specific stuff in every single page.  Don't know though how hard could be to implement from panda side.

**Reply by astahmer** [reply]
this shouldnt be too hard, I'll be looking at this very soon !

**Reply by astahmer** [reply]
here it is https://github.com/chakra-ui/panda/pull/1448

**Reply by Emiliano-Bucci** [reply]
@astahmer Hi! I see that the pr is not yet merged; is there a way to locally test the implementation? I've tried to install panda from the branch but it seems it doesn't work

**Reply by astahmer** [reply]
- clone panda repo, go into `feat/cssgen-minimal-and-of-type` branch, run `pnpm i --ignore-scripts` in the monorepo root
- inside your app, run `pnpm add @pandacss/dev@link:/absolute/path/to/panda/packages/cli` and then you should be able to use the `pnpm panda cssgen static` command
